### ISSUE TITLE
Peripheral disconnect recognition

### ISSUE THREAD

**Issue by quanguyen** [jordanebelanger/SwiftyBluetooth#38]
Hi,

How should I detect a peripheral disconnected from the master?

Thanks,

**Comment by komaxx** [comment]
Same issue here: How can I monitor an already established connection?

**Comment by jordanebelanger** [comment]
> Same issue here: How can I monitor an already established connection?

Hi @komaxx, sorry to hear you're having issues, I've been getting overloaded with work, notably server side Swift stuff and haven't had much time to work on this library.

Do you know of any specific CoreBluetooth hook/notification that does what you need in this case and if so could you share a link to the documentation so I can take a look?

If it's not too hard, you may even want to consider doing a PR for it and I will gladly merge it.

**Comment by adamontherun** [comment]
@jordanebelanger I have the same issue. I think the fix would be to expose the centralManager(_:didDisconnectPeripheral:error:) delegate method via your framework.

**Comment by sirowain** [comment]
Any updates on this request? I'm also interested.

**Comment by jordanebelanger** [comment]
@sirowain https://github.com/jordanebelanger/SwiftyBluetooth/releases/tag/2.2.0

**Comment by sirowain** [comment]
This is not always working, the method returns before calling the postCentralEvent if you don't have any disconnectRequest for the uuid. If you are trying to detect an unexpected peripheral disconnect you'll miss it.
```
let uuid = peripheral.identifier
guard let request = disconnectRequests[uuid] else {
    return
}
...
postCentralEvent(Central.CentralCBPeripheralDisconnected, userInfo: userInfo)
```

**Comment by jordanebelanger** [comment]
> This is not always working, the method returns before calling the postCentralEvent if you don't have any disconnectRequest for the uuid. If you are trying to detect an unexpected peripheral disconnect you'll miss it.
> 
> ```
> let uuid = peripheral.identifier
> guard let request = disconnectRequests[uuid] else {
>     return
> }
> ...
> postCentralEvent(Central.CentralCBPeripheralDisconnected, userInfo: userInfo)
> ```

Good catch, I made this very fast yesterday.

A pr with the fix would be appreciated.

**Comment by jordanebelanger** [comment]
@sirowain Fixed in 2.2.1
